tree: Example undo flow with diagrams (#14533)
microsoft/FluidFramework: packages/dds/tree/docs/undo/v1-undo-example-flow.md

# V1 Undo Example Flow

**Legend**:

> ```{mermaid}

Reviewer: I think these need to be `mermaid` and not `{mermaid}` for the diagrams to embed properly ([github's docs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams#creating-mermaid-diagrams)). When I view the rich diff I'm seeing the "code" for the diagram instead of the diagrams themselves.

flowchart
    remote{{Denotes commits by a remote session}}
    local[Denotes commits by the local session]
```

> This example starts with a trunk and single local branch with a few normal commits and a single undo commit. The [undo commit tree](./v1-undo.md#the-undo-commit-tree) contains `A1` and `B2`. The head undo commit for the trunk branch is `A1` because it is the latest local edit on the trunk while `B2` is the head undo commit for the local branch. There is nothing in the [redo commit tree](./v1-undo.md#the-redo-commit-tree) because a normal edit, `B2`, clears the redo-able edits.

Reviewer 2: When writing markdown docs, please consider using lazy returns. They make code reviews easier by making it less likely that multiple independent comments will need to be made for the same line, and they also make suggestions like this one shorter:

```suggestion
This example starts with a trunk and single local branch with a few normal commits and a single undo commit.
The [undo commit tree](./v1-undo.md#the-undo-commit-tree) contains `A1` and `B2`.
The head undo commit for the trunk branch is `A1` because it is the latest edit from the local session (and has not been undone).
`B2` is the head undo commit for the local branch.
There is nothing in the [redo commit tree](./v1-undo.md#the-redo-commit-tree) because a normal edit, `B2`, clears the redo-able edits.
```
